tests/mac_minidump_writer.rs

```diff
@@ -0,0 +1,194 @@
+        // to whole seconds. And it seems that crashpad doesn't have tests around
+        // this, though that's hard to say given how tedious it is finding stuff
+        // in that bloated codebase
+        // assert!(mi.process_user_time > 0);
+        // assert!(mi.process_kernel_time > 0);
+
+        // These aren't currently available on aarch64, or if they are, they
+        // are not via the same sysctlbyname mechanism. Would be nice if Apple
+        // documented...anything
+        if cfg!(target_arch = "x86_64") {
+            assert!(mi.processor_max_mhz > 0);
+            assert!(mi.processor_current_mhz > 0);
+        }
+    } else {
+        panic!("unexpected misc info type {:?}", misc_info);
+    }
+}
+
+// /// Validates we can actually walk the stack for each thread in the minidump,
+// /// this is using minidump-processor, which (currently) depends on breakpad
+// /// symbols, however https://github.com/mozilla/dump_syms is not available as
+// /// a library https://github.com/mozilla/dump_syms/issues/253, so we just require
+// /// that it already be installed, hence the ignore
+// #[test]
+// #[ignore = "ignored, requires dump_syms installed"]
+// fn stackwalks() {
+//     println!("generating minidump...");
+//     let md = capture_minidump("stackwalks", mach2::exception_types::EXC_BREAKPOINT);
+
+//     // Generate the breakpad symbols
+//     println!("generating symbols...");
+//     let mut cmd = std::process::Command::new("dump_syms");
+//     cmd.args(["-s", ".test-symbols", "target/debug/test"]);
+//     assert!(cmd.status().unwrap().success());
+
+//     let provider =
+//         minidump_processor::Symbolizer::new(minidump_processor::simple_symbol_supplier(vec![
+//             ".test-symbols".into(),
+//         ]));
+
+//     let state = futures::executor::block_on(async {
+//         minidump_processor::process_minidump(&md.minidump, &provider).await
+//     })
+//     .unwrap();
+
+//     //state.print(&mut std::io::stdout()).map_err(|_| ()).unwrap();
+
+//     // We expect at least 2 threads, one of which is the fake crashing thread
+//     let fake_crash_thread = state
+//         .threads
+//         .iter()
+//         .find(|cs| cs.thread_id == md.thread)
+//         .expect("failed to find crash thread");
+
+//     // The thread is named, however we currently don't retrieve that information
+//     // currently, indeed, it appears that you need to retrieve the pthread that
+//     // corresponds the mach port for a thread, however that API seems to be
+//     // task specific...
+//     // assert_eq!(
+//     //     fake_crash_thread.thread_name.as_deref(),
+//     //     Some("test-thread")
+//     // );
+
+//     assert!(
+//         fake_crash_thread.frames.iter().any(|sf| {
+//             sf.function_name
+//                 .as_ref()
+//                 .map_or(false, |fname| fname.ends_with("wait_until_killed"))
+//         }),
+//         "unable to locate expected function"
+//     );
+// }
```

Comment:
This is temporarily disabled, but I think this would be a good way to test all the platforms, making the test binary more complicated, and verify that minidump-process finds the expected threads and stackframes. As noted, the dump_syms issue where it is only packaged as a binary right now makes it annoying to use in CI at the moment, and breakpad symbols depends on openssl which unnecessarily complicates builds, especially since this is all local symbol resolution. (at least for now)